cutil/retry/retry.go

```diff
@@ -0,0 +1,54 @@
+// Copyright © 2017 The Kubicorn Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package retry
+
+import (
+	"os"
+	"time"
+
+	"github.com/kris-nova/kubicorn/cutil/signals"
+)
+
+type Retry interface {
+	Work()
+}
+
+type Worker struct {
+	Function func()
```

Comment:
So this limits our `Worker` to only being able to call a function.. if we could implement an interface here instead of `Retry` we could define our own implementation.. and then the `Worker` struct could just call `interface.Run()` for instance.. This is probably the worst way of explaining it ever.. but if you need me to clarify just ping me